### DISCUSSION TITLE
Plot size (Figure size)

### DISCUSSION THREAD

**Discussion by bilgetutak** [OpenDrift/opendrift#956] [Q&A] [answered]
Hi All,
It seems the plot function in the run class does not have a way of defining the plot size. The only possible solution I found is to get the return values from

`ax, plt = o.plot()`

then obtain the current figure using

`fig = plt.gcf()`

Then finally set the width and height through

`fig.set(figwidth=w, figheight=h)`

Is there a better solution for the figure size setting.

If not, maybe it can be implemented in the plot or set_up_map function parameters (I can implement)? What do you think?
Thanks in advance,
Bilge.

**Comment by knutfrode** [comment, marked as answer]
Hi,

The aspect ratio is calculated automatically from actual extent in `set_up_map`,
and figure size is the automatically 11 inches along the longest axis, and 11 inches * aspect_ratio along the shortest axis:
https://github.com/OpenDrift/opendrift/blob/master/opendrift/models/basemodel.py#L3238

If this automatic aspect ratio is not used, your figure will become distorted. However, the size of 11 inches could in principle be made an input parameter, if that would help in your case?

**Reply by bilgetutak** [reply]
Hi @knutfrode   Thanks for the answer.
I have found the aspect ration related calculation, and because of the geographic mapping using the aspect ratio is logical. However, for my application 11 inches is quite large and when I am playing with the model, I want to plot much smaller figures. I think making the 11 inch a settable parameter might things simpler (at least for my case).

**Comment by knutfrode** [comment]
I now added `figsize` as an optional input parameter to `plot` and `animation`methods: https://github.com/OpenDrift/opendrift/pull/957

Default is 11 inches as before.
